core/net/rpl/rpl-mrhof.c

```diff
@@ -118,6 +120,23 @@ reset(rpl_dag_t *dag)
 }
 
 static void
+dao_ack_callback(rpl_parent_t *p, int status)
+{
+  if(status == RPL_DAO_ACK_UNABLE_TO_ACCEPT_ROOT) {
+    return;
+  }
+  /* here we need to handle failed DAO's and other stuff */
+  PRINTF("RPL: MRHOF - DAO ACK received with status: %d\n", status);
+  if(status >= RPL_DAO_ACK_UNABLE_TO_ACCEPT) {
+    /* punish the ETX as if this was 10 packets lost */
+    neighbor_link_callback(p, MAC_TX_OK, 10);
```

Comment:
Ideally I think this shouldn't affect link metrics, but rather be recorded as a different flag. This way we can treat differently two different problems: broken links and parents unable to have us as a child. Could be addressed in a different PR.